lib/cloud-watch-event-listener.js

```diff
@@ -104,15 +104,102 @@ class CloudWatchEventListener extends events.EventEmitter {
     let region = body.region;
     let id = body.detail['instance-id'];
     let state = body.detail.state;
+    // Not a great name, but this variable is the time that the CloudWatch
+    // Event message was generated by the API, not the time at which we
+    // received it
+    let generated = new Date(body.time);
 
     try {
-      await this.state.logCloudWatchEvent({region, id, state});
+      await this.state.logCloudWatchEvent({region, id, state, generated});
     } catch (err) {
-      // We don't want to block things
+      // We don't want to block things, but let's bubble up the error for the
+      // time being
       this.monitor.reportError(err);
     }
 
-    if (state === 'pending' || state === 'running') {
+    let dbResponse;
+
+    let transaction = await this.state.beginTransaction();
+
+    try {
+      let x = await this.state.listInstances({region, id}, transaction);
+      assert(Array.isArray(x));
```

Comment:
not really, but I prefer to over-assert so that errors are detected as early a call chain as possible